WordPress/Classes/ViewRelated/Media/Tenor/TenorMedia.swift

```diff
@@ -0,0 +1,123 @@
+import MobileCoreServices
+import WPMediaPicker
+
+struct TenorImageCollection {
+    let largeURL: URL
+    let previewURL: URL
+    let staticThumbnailURL: URL
+    let largeSize: CGSize
+}
+
+// Models a Tenor image
+
+final class TenorMedia: NSObject {
+    let id: String
+    let name: String
+    let caption: String
+    let updatedDate: Date
+    let images: TenorImageCollection
+
+    init(id: String, name: String, caption: String, images: TenorImageCollection, date: Date? = nil) {
+        self.id = id
+        self.name = name
+        self.caption = caption
+        self.updatedDate = date ?? Date()
+        self.images = images
+    }
+}
+
+// MARK: - Create Tenor media from API GIF Entity
+
+extension TenorMedia {
+    convenience init?(tenorGIF gif: TenorGIF) {
+        let largeGif = gif.media.first { $0.gif != nil }?.gif
```

Comment:
I could be misunderstanding what's happening here, but it seems like there's (probably a slim) chance the 3 gifs could be different. For each gif, you're getting the first media where that gif exists. It seems to me we should get the first media (or at least, the same one), then check the 3 exist on that one.

Reply:
Yeah, that looks confusing, however, as the TenorAPI returns a `TenorGIF` which contains an array of `TenorGIFCollection`, and in each `TenorGIFCollection` there will be only one key has value. To avoid various `TenorGIFTinyMedia`, `TenorGIFMediumMedia`, etc. I chose to put all of `tiny`, `medium` as optional values in `TenorGIFCollection` so the code is much shorter.